src/anndata/experimental/merge.py

```diff
@@ -100,27 +100,34 @@ def _gen_slice_to_append(
 
 
 @singledispatch
-def as_group(store, *args, **kwargs) -> ZarrGroup | H5Group:
+def as_group(store, *, mode: str) -> ZarrGroup | H5Group:
     msg = "This is not yet implemented."
     raise NotImplementedError(msg)
 
 
 @as_group.register(os.PathLike)
 @as_group.register(str)
-def _(store: os.PathLike | str, *args, **kwargs) -> ZarrGroup | H5Group:
+def _(store: os.PathLike | str, *, mode: str) -> ZarrGroup | H5Group:
     store = Path(store)
     if store.suffix == ".h5ad":
         import h5py
 
-        return h5py.File(store, *args, **kwargs)
-    import zarr
+        return h5py.File(store, mode=mode)
 
-    return zarr.open_group(store, *args, **kwargs)
+    if mode == "r":  # others all write: r+, a, w, w-
+        import zarr
+
+        return zarr.open_group(store, mode=mode)
+
+    from anndata._io.zarr import open_write_group
+
+    return open_write_group(store, mode=mode)
 
 
 @as_group.register(ZarrGroup)
 @as_group.register(H5Group)
-def _(store, *args, **kwargs):
+def _(store, *, mode: str) -> ZarrGroup | H5Group:
```

Comment:
codecov says this is never hit. seems weird!

Reply:
Was that way before then, it would seem.  It's possible no one was testing `concat_on_disk` with an actual store